app/WebParsing/ReqParser.hs

```diff
@@ -92,31 +100,47 @@ gradeParser = do
 -- parse for cutoff percentage before a course
 coBefParser :: Parser Req
 coBefParser = do
-    _ <- Parsec.choice $ map (Parsec.try . (>> Parsec.space) . Parsec.string) ["minimum grade of", "minimum mark of", "minimum of", "minimum"]
-    Parsec.spaces
-    grade <- gradeParser
+    _ <- Parsec.optional $ caseInsensitiveStr "an " <|> caseInsensitiveStr "a "
```

Comment:
better to remove the space here and still use `Parsec.spaces` (as before). This will allow for parsing multiple spaces (put in accidentally, for example).

Reply:
I put the space there because I want to prevent the letter "a" from being consumed if it is part of another word. For example, we might have prereqs in the form of "At least C+ in courseCode". Having that space seems like it is the simplest way to prevent the A from being consumed (Consuming the "A" would cause issues with `cutoffHelper` when trying to parse the grade by looking for the key word "At least").

In order to catch multiple spaces following "An" or "A", I put another call to `Parsec.spaces` after this line.